File: python3/1700_number_of_students_unable_to_eat_lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        counts = [0, 0]
        
        for s in students:
            counts[s] += 1

        for s in sandwiches:
            counts[s] -= 1

            if counts[s] < 0:
                return counts[1 - s]

        return 0

File: rust/1700_number_of_students_unable_to_eat_lunch.rs
impl Solution {
  pub fn count_students(students: Vec<i32>, sandwiches: Vec<i32>) -> i32 {
    let mut counts = [0, 0].to_vec();

    for s in students {
      counts[s as usize] += 1;
    }

    for s in sandwiches {
      counts[s as usize] -= 1;

      if counts[s as usize] < 0 {
        return counts[1 - s as usize];
      }
    }

    0
  }
}
